src/libs/tagging.c

```diff
@@ -384,12 +536,230 @@ static void delete_button_clicked(GtkButton *button, gpointer user_data)
   }
   g_list_free(g_list_first(tagged_images));
 
-  update(self, 0);
-  update(self, 1);
+  dt_control_signal_raise(darktable.signals, DT_SIGNAL_TAG_CHANGED);
+}
+
+static void rename_button_clicked(GtkButton *button, gpointer user_data)
+{
+  dt_lib_module_t *self = (dt_lib_module_t *)user_data;
+  dt_lib_tagging_t *d = (dt_lib_tagging_t *)self->data;
+
+  char *tagname;
+  gchar *text;
+  GtkWidget *label;
+  GtkTreeIter iter;
+  GtkTreeModel *model = NULL;
+  GtkTreeView *view = d->related;
+  GtkTreeSelection *selection = gtk_tree_view_get_selection(GTK_TREE_VIEW(view));
+  if(!gtk_tree_selection_get_selected(selection, &model, &iter)) return;
+
+  if (model == GTK_TREE_MODEL(d->liststore))
+    gtk_tree_model_get(model, &iter, DT_LIB_TAGGING_COL_TAG, &tagname, -1);
+  else
+    gtk_tree_model_get(model, &iter, DT_LIB_TAGGING_TREE_COL_PATH, &tagname, -1);
+  char *subtag = g_strrstr(tagname, "|");
+
+  gint tag_count;
+  gint img_count;
+  dt_tag_count_tags_images(tagname, &tag_count, &img_count);
+  if (tag_count == 0) return;
+
+  GtkWidget *win = dt_ui_main_window(darktable.gui->ui);
+  GtkWidget *dialog = gtk_dialog_new_with_buttons("Rename tag?", GTK_WINDOW(win), GTK_DIALOG_DESTROY_WITH_PARENT,
```

Comment:
The title should be in lower-case and translatable.

Reply:
ok